k2/python/tests/ctc_gradients_test.py

```diff
@@ -36,20 +37,43 @@ def build_ctc_topo(tokens: List[int]) -> k2.Fsa:
 
     num_states = len(tokens)
     final_state = num_states
-    rules = ''
+    arcs = ''
     for i in range(num_states):
         for j in range(num_states):
             if i == j:
-                rules += f'{i} {i} 0 {tokens[i]} 0.0\n'
+                arcs += f'{i} {i} {tokens[i]} 0 0.0\n'
             else:
-                rules += f'{i} {j} {tokens[j]} {tokens[j]} 0.0\n'
-        rules += f'{i} {final_state} -1 -1 0.0\n'
-    rules += f'{final_state}'
-    ans = k2.Fsa.from_str(rules)
-    ans = k2.arc_sort(ans)
+                # we assume that eps has the same ID, i.e., 0, with
```

Comment:
I suppose we can delete these comments with above `CAUTION`?

Reply:
fixed.